src/VS.Mefx/ConfigCreator.cs

```diff
@@ -0,0 +1,491 @@
+                    }
+                }
+            }
+
+            this.OutputCacheFile = options.CacheFile;
+        }
+
+        /// <summary>
+        /// Gets or sets the command line arguments specified by the user.
+        /// </summary>
+        private CLIOptions Options { get; set; }
+
+        /// <summary>
+        /// Gets the catalog that stores information about the imported parts.
+        /// </summary>
+        public ComposableCatalog? Catalog { get; private set; }
+
+        /// <summary>
+        /// Gets configuration information associated with the imported parts.
+        /// </summary>
+        public CompositionConfiguration? Config { get; private set; }
+
+        /// <summary>
+        /// Gets or sets a dictionary storing parts indexed by thier parts name for easy lookup.
+        /// </summary>
+        public Dictionary<string, ComposablePartDefinition> PartInformation { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path of the cache file to store the processed parts.
+        /// </summary>
+        private string? OutputCacheFile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the paths to the assembly files we want to read.
+        /// </summary>
+        private List<string> AssemblyPaths { get; set; }
+
+        /// <summary>
+        /// Gets or sets the paths to the cache files we want to read.
+        /// </summary>
+        private List<string> CachePaths { get; set; }
+
+        /// <summary>
+        /// Method to get the details about a part, i.e. the part Definition, given its name.
+        /// </summary>
+        /// <param name="partName"> The name of the part we want to get details about.</param>
+        /// <returns><see cref="ComposablePartDefinition"/> associated with the given part if it is
+        /// present in the catalog and null otherwise.</returns>
+        public ComposablePartDefinition? GetPart(string partName)
+        {
+            if (!this.PartInformation.ContainsKey(partName))
+            {
+                return null;
+            }
+
+            return this.PartInformation[partName];
+        }
+
+        /// <summary>
+        /// Method to intialize the catalog and configuration objects from the input files.
+        /// </summary>
+        /// <returns>A Task object when all the assembly have between loaded in and configured.</returns>
+        public async Task Initialize()
+        {
+            CustomAssemblyLoader customLoader = new CustomAssemblyLoader(this.Options.ErrorWriter);
+            Resolver customResolver = new(customLoader);
+            var nugetDiscover = new AttributedPartDiscovery(customResolver, isNonPublicSupported: true);
+            var netDiscover = new AttributedPartDiscoveryV1(customResolver);
+            PartDiscovery discovery = PartDiscovery.Combine(customResolver, netDiscover, nugetDiscover);
+
+            if (this.AssemblyPaths.Count() > 0)
```

Comment:
`Count()` is an extension method that will enumerate every element in the collection to count them. But `List<T>.Count` is a property that can readily return the answer, and would be the most performant option in this case.

When `Count` is *not* available as a property, use `Any()` instead of `Count() > 0` because it will only enumerate at most one element.